pkg/runtime/wfengine/workflow.go

```diff
@@ -521,6 +526,10 @@ func (wf *workflowActor) loadInternalState(ctx context.Context, actorID string)
 		// No such state exists in the state store
 		return nil, nil
 	}
+	if !wf.cachingDisabled {
+		// update cached state
+		wf.states.Store(actorID, state)
+	}
```

Comment:
do we need the same check at the start of the function, at line 514?

Reply:
Also in `saveInternalState` the cache is being updated even if the TransactionStateOperation fails lines 539 and 549?

Reply:
Not strictly necessary for 514 since the cache should be empty.

For 539, 549, yes this needs to be fixed. That issue is why some Cosmos DB compatibility problems weren’t more quickly detected. Could do it now or in another PR.